src/api/artifact/model.go

```diff
@@ -50,13 +50,14 @@ type ResourceLink struct {
 type Option struct {
```

Comment:
We have already had so many options, again, maybe the function option is more suitable now.

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis